### DISCUSSION TITLE
implement Korea's ESS expansion model

### DISCUSSION THREAD

**Discussion by RogerKwak** [pypsa-meets-earth/pypsa-earth#967] [Q&A] [answered]
Hi, everyone. Thanks for these great creature.

I would like to implement Korea's ESS expansion model through PyPSA - earth.
My plan is below.

1. Decompose the .nc file into a .csv file and enter Korean data.
2. Implement maps and grids like PyPSA-Eur

Is this really possible?

Or does PyPSA-earth already have Korean data?

**Comment by davide-f** [comment, marked as answer]
Hello @RogerKwak :D
It is great to have you posting here :)

PyPSA-Earth already provides you with a representative model of Korea, built with our standard assumptions.
A validation for each country is however appropriate to check the validity of the model.

On how to do that, the first step would be to run pypsa-earth for Korea :)
After the installation and the short tutorial, the procedure is explained here:
https://pypsa-earth.readthedocs.io/en/latest/tutorial.html

Then with also the notebooks in our documentation package, you can explore the output model and see if changes are necessary:
https://pypsa-earth.readthedocs.io/en/latest/notebooks.html

If you have local data, you can definitely include your own data while building the model by using the options in the option "clean_osm_data_options" in the config file, where you see the suboptions custom_path_{object} and use_custom_{object}.
However, that may not be necessary hopefully :D

At the following link, you can get an idea of the network topology you can get (maps about 1y old) for any UN country worldwide:
https://zenodo.org/records/10080766

:)

**Reply by RogerKwak** [reply]
Thank you for your kindness reply.

I'll try it first, and if it doesn't work or there's something I don't know, I'll ask another question.

**Reply by davide-f** [reply]
Write as you please :D
We also have a discord channel for more interactions and were you may be able to find some questions already answered.
You are welcome to join :)
Link: https://discord.gg/AnuJBk23FU
